### PR TITLE
[RND-334] authenticate to request a token

### PR DESCRIPTION
**Pull request by bradbanister**

### CODE REVIEW

File: Meadowlark-js/backends/meadowlark-mongodb-backend/src/repository/authorization/GetAuthorizationClient.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    if (result === null) return { response: 'GET_FAILURE_NOT_EXISTS' };
+
+    return {
+      response: 'GET_SUCCESS',
+      clientName: result.clientName,
+      roles: result.roles,
+      clientSecretHashed: result.clientSecretHashed,
+    };
+  } catch (e) {
+    return { response: 'UNKNOWN_FAILURE' };
```
Comment (stephenfuqua) [inline Meadowlark-js/backends/meadowlark-mongodb-backend/src/repository/authorization/GetAuthorizationClient.ts:28]: Should this be logging the error?

File: Meadowlark-js/packages/meadowlark-authz-server/src/handler/RequestToken.ts [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+  }
+  return Buffer.from(signingKeyEncoded, 'base64');
+}
+const cachedSigningKey = memoize(signingKey);
+
+/*
+ * Creates a standard Meadowlark Jwt.
+ */
+function createToken(clientId: string, vendor: string, roles: string[]): Jwt {
+  const claims = { iss: 'ed-fi-meadowlark', aud: 'meadowlark', roles, client_id: clientId };
```
Comment (stephenfuqua) [inline Meadowlark-js/packages/meadowlark-authz-server/src/handler/RequestToken.ts:40]: I wonder if these should be config values?

File: Meadowlark-js/packages/meadowlark-authz-server/src/handler/RequestToken.ts [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+
+  const token: Jwt = createJwt({ ...claims, sub: vendor }, cachedSigningKey()) as Jwt;
+
+  token.setExpiration(new Date().getTime() + 60 * 60 * 1000); // One hour from now
```
Comment (stephenfuqua) [inline Meadowlark-js/packages/meadowlark-authz-server/src/handler/RequestToken.ts:44]: Perhaps the length of time should be a config value? (e.g. # of milliseconds)

File: Meadowlark-js/backends/meadowlark-mongodb-backend/src/repository/authorization/GetAuthorizationClient.ts (hunk 1 above)
Comment (bradbanister) [reply Meadowlark-js/backends/meadowlark-mongodb-backend/src/repository/authorization/GetAuthorizationClient.ts:28]: Will do

File: Meadowlark-js/packages/meadowlark-authz-server/src/handler/RequestToken.ts (hunk 2 above)
Comment (stephenfuqua) [reply Meadowlark-js/packages/meadowlark-authz-server/src/handler/RequestToken.ts:40]: I'm going to create a separate ticket that will cover:

1. moving these to config values
2. documenting the oauth server setup

File: Meadowlark-js/packages/meadowlark-authz-server/src/handler/RequestToken.ts (hunk 3 above)
Comment (stephenfuqua) [reply Meadowlark-js/packages/meadowlark-authz-server/src/handler/RequestToken.ts:44]: Not in this ticket

**Comment by stephenfuqua** [comment]
How do I bootstrap the initial admin user now?

**Comment by bradbanister** [comment]
Forgot to commit log fix and updated .http test file. New token request endpoint is "local/oauth/token", hard coded credentials are still in place and checked first.

**Review by stephenfuqua — APPROVED** [review]

**Pull request merged**